### PR TITLE
Create failing tests dropdown component in dashboard tab summary

### PR DESCRIPTION
**Pull request by bryan-lou**

### CODE REVIEW

File: web/src/testgrid-tab-table.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { LitElement, html, css } from 'lit';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { customElement, property } from 'lit/decorators.js';
+import {TabSummaryInfo} from './testgrid-dashboard-summary';
+
+@customElement('testgrid-tab-table')
+export class TestgridTabTable extends LitElement {
+  @property() clicked = '';
+  @property() visible = false;
```
Comment (sultan-duisenbay) [inline web/src/testgrid-tab-table.ts:9]: this should probably be a state and not a property. please refer to https://lit.dev/docs/components/properties/#internal-reactive-state

File: web/src/testgrid-tab-table.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  info?: TabSummaryInfo;
```
Comment (sultan-duisenbay) [inline web/src/testgrid-tab-table.ts:10]: For this particular element we don't need the full tabSummaryInfo, right? We can just provide failuresSummary

File: web/src/testgrid-tab-table.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+              <th>Last Passed</th>
+            </tr>
+          </table>`
+          : ''}
+      </div>
+    `
+  }
+  private dropdownTable(){
+    this.visible = !this.visible;
+    this.dispatchEvent(new CustomEvent('visibleChange', { detail: this.visible }));
```
Comment (sultan-duisenbay) [inline web/src/testgrid-tab-table.ts:33]: what is the purpose of this event?

**Review by sultan-duisenbay — CHANGES_REQUESTED** [review]
Thanks for the PR, Bryan! Left comments here and there. The major one is to finalize a separate element for the table data and embed it into `tab-summary` element.

In addition to this, would be great to add it to tab summary stories - make some of them have failuresSummary. Also add the test cases for tab summary

File: web/src/tab-summary.ts [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import { LitElement, html, css } from 'lit';
+import { map } from 'lit/directives/map.js';
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 import { customElement, property } from 'lit/decorators.js';
 import { TabSummaryInfo } from './testgrid-dashboard-summary';
 @customElement('tab-summary')
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export class TabSummary extends LitElement {
   @property({ type: Object })
+  @property() visible = false;
```
[the remark this answers is not in the document]
Comment (bryan-lou) [reply web/src/tab-summary.ts:10]: done

File: web/src/testgrid-dashboard-summary.ts [hunk 5]
```diff
@@ -42,6 +58,27 @@ function convertResponse(ts: TabSummary) {
     latestGreenBuild: ts.latestPassingBuild,
     dashboardName: ts.dashboardName,
   };
+  if (ts.failuresSummary !== undefined) {
+    tsi.failuresSummary = {} as FailuresSummaryInfo
+  }
+  if (ts.failuresSummary?.failureStats !== undefined) {
```
[the remark this answers is not in the document]
Comment (bryan-lou) [reply web/src/testgrid-dashboard-summary.ts:64]: done

File: web/src/testgrid-tab-table.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { LitElement, html, css } from 'lit';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { customElement, property } from 'lit/decorators.js';
+import {TabSummaryInfo} from './testgrid-dashboard-summary';
+
+@customElement('testgrid-tab-table')
+export class TestgridTabTable extends LitElement {
+  @property() clicked = '';
```
[the remark this answers is not in the document]
Comment (bryan-lou) [reply web/src/testgrid-tab-table.ts:8]: removed

File: web/src/testgrid-tab-table.ts (hunk 2 above)
Comment (bryan-lou) [reply web/src/testgrid-tab-table.ts:10]: Yup, that's right, but actually right now I imported the `TabSummaryInfo`, but there's no data in this object when it gets called. So I'm still looking into this and trying to figure out why it's working in tab-summary but not this

File: web/src/testgrid-tab-table.ts (hunk 1 above)
Comment (bryan-lou) [reply web/src/testgrid-tab-table.ts:9]: why shouldn't we use a property for this?

File: web/src/testgrid-tab-table.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import { LitElement, html, css } from 'lit';
```
[the remark this answers is not in the document]
Comment (sultan-duisenbay) [reply web/src/testgrid-tab-table.ts:1]: testgrid-tab-table

File: web/src/testgrid-tab-table.ts (hunk 1 above)
Comment (sultan-duisenbay) [reply web/src/testgrid-tab-table.ts:8]: property is an externally set attribute while state is internal. looks like visible is toggled within the table element

File: web/src/tab-summary.ts [hunk 8]
```diff
@@ -39,11 +41,37 @@ export class TabSummary extends LitElement {
           </div>
         </div>
       </div>
+      ${this.info?.failuresSummary !== undefined ? html `<div class="dropdown-container">
```
Comment (sultan-duisenbay) [inline web/src/tab-summary.ts:44]: would be great to ember the table element you defined below here

**Review by sultan-duisenbay — CHANGES_REQUESTED** [review]

File: web/src/testgrid-tab-table.ts (hunk 1 above)
Comment (bryan-lou) [reply web/src/testgrid-tab-table.ts]: fixed

File: web/src/testgrid-tab-table.ts (hunk 2 above)
Comment (bryan-lou) [reply web/src/testgrid-tab-table.ts]: Done

File: web/src/tab-summary.ts (hunk 8 above)
Comment (bryan-lou) [reply web/src/tab-summary.ts]: done

File: web/src/testgrid-failures-summary.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import { LitElement, html, css } from 'lit';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { customElement, property, state } from 'lit/decorators.js';
+import { TabSummaryInfo } from './testgrid-dashboard-summary';
+import { map } from 'lit/directives/map.js';
+
+@customElement('testgrid-failures-summary')
+export class TestgridTabTable extends LitElement {
```
Comment (sultan-duisenbay) [inline web/src/testgrid-failures-summary.ts:8]: It's better to keep the class name the same the custom element name. i.e
```
export class TestgridFailuresSummary extends LitElement
```

**Review by sultan-duisenbay — CHANGES_REQUESTED** [review]
One nit with changing class name

File: web/src/testgrid-failures-summary.ts (hunk 9 above)
Comment (bryan-lou) [reply web/src/testgrid-failures-summary.ts]: Done!

**Review by sultan-duisenbay — APPROVED** [review]
/hold

Please squash the commits (have 1 instead of 7)

**Pull request merged**